agent/tool-scripts/kvm-spinlock

```diff
@@ -235,6 +236,55 @@ case "$tool" in
 	tool_cmd="$script_path/datalog/$tool-datalog $interval $username $password $tool_stdout_file"
 	;;
 esac
+
+function safe_kill() {
+	# safer kill: check for strange situations and deal with them.
+	# If there is a pid, try to kill it.  If the kill succeeds,
+	# return success, but if it fails, try to see if the pid is
+	# still running: if so, try to kill it with a KILL and return
+	# failure.  Eat any error message.
+	local tool=$1
+	local pid=$2
+	local signal${3:-TERM}
+	local pidlist p
+
+	if [ -z "$pid" ] ;then
+		# should not happen
+		return 1
+	fi
+	
+	# check that the pid corresponds to the tool
+	pidlist=$(pidof $tool)
```

Comment:
If `pidof` returns an empty string because `$tool` is not running, I think the following `for p in $pidlist ;do` will fail with a bash syntax error.